vortex-duckdb/src/convert/array/mod.rs

```diff
@@ -103,24 +168,20 @@ pub fn selection_vector_from_slice<P: NativePType + AsPrimitive<u32>>(
     slice.iter().map(|v| (*v).as_()).collect()
 }
 
-const ALL_TRUE_SEL_MASK: [u64; 32] = [u64::MAX; 32];
-const ALL_FALSE_SEL_MASK: [u64; 32] = [u64::MIN; 32];
-
 pub fn write_validity_from_mask(mask: Mask, flat_vector: &mut FlatVector) {
     // Check that both the target vector is large enough and the mask too.
     // If we later allow vectors larger than 2k (against duckdb defaults), we can revisit this.
-    assert!(mask.len() <= DUCKDB_STANDARD_VECTOR_SIZE);
-    assert!(flat_vector.capacity() <= DUCKDB_STANDARD_VECTOR_SIZE);
+    assert!(mask.len() <= flat_vector.capacity());
     match mask {
         Mask::AllTrue(len) => {
             if let Some(slice) = flat_vector.validity_slice() {
                 // This is only needed if the vector as previously allocated.
-                slice.copy_from_slice(&ALL_TRUE_SEL_MASK[0..len]);
+                slice[0..len].fill(u64::MAX)
             }
         }
         Mask::AllFalse(len) => {
             let slice = flat_vector.init_get_validity_slice();
-            slice.copy_from_slice(&ALL_FALSE_SEL_MASK[0..len])
+            slice[0..len].fill(u64::MIN)
```

Comment:
Is `[0..len]` different here from `[..]`, aka get a slice to the whole vector?

Reply:
If so, let's go with that.

Reply:
yes, the slice len is the vector capacity, but the len is the mask length, which are not equa